operator/duties/dutystore/duties.go

```diff
@@ -96,3 +96,11 @@ func (d *Duties[D]) ResetEpoch(epoch phase0.Epoch) {
 
 	delete(d.m, epoch)
 }
+
+func (d *Duties[D]) IsEpochSet(epoch phase0.Epoch) bool {
+	d.mu.RLock()
+	defer d.mu.RUnlock()
+
+	_, exists := d.m[epoch]
+	return exists
+}
```

Comment:
this is risky because there are conditions where this will never be set, such as previous epoch, when failing to fetch duties or when (wrongly) detecting 0 active validators in the network

Reply:
maybe we could ignore the previous epoch issue if we delete epoch-2 instead of epoch-1 in the handlers (should be ok since msg validation should stop msgs for those epochs for 'too late')

and just ensure epoch is always set, even if to empty, after handler processed slot (despite failures)

wdyt?

Reply:
solved